apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -676,15 +676,121 @@ function openLinksInParentFrame( calypsoPort ) {
 		} );
 	} );
 
-	if ( calypsoifyGutenberg.manageReusableBlocksUrl ) {
-		const manageReusableBlocksLinkSelectors = [
-			'.block-editor-inserter__manage-reusable-blocks', // Link in the Blocks Inserter
-			'a.components-menu-item__button[href*="post_type=wp_block"]', // Link in the More Menu
-		].join( ',' );
-		$( '#editor' ).on( 'click', manageReusableBlocksLinkSelectors, ( e ) => {
-			e.preventDefault();
-			window.open( calypsoifyGutenberg.manageReusableBlocksUrl, '_top' );
+	const { createNewPostUrl, manageReusableBlocksUrl } = calypsoifyGutenberg;
+	if ( ! createNewPostUrl && ! manageReusableBlocksUrl ) {
+		return;
+	}
+
+	await isEditorReadyWithBlocks();
+
+	// Create a new post link in block settings sidebar for Query block
+	const createNewPostLinkObserver = new window.MutationObserver( () => {
+		const hyperlink = document.querySelector( '.wp-block-query__create-new-link a' );
+		if ( hyperlink ) {
+			hyperlink.href = createNewPostUrl;
+			hyperlink.target = '_top';
+		}
+	} );
+
+	// Manage reusable blocks link in the global block inserter's Reusable tab
+	// Post editor only
+	const inserterManageReusableBlocksObserver = new window.MutationObserver( ( mutations ) => {
+		const node = mutations[ 0 ].target;
+		if ( node.ariaSelected === 'true' ) {
+			const hyperlink = document.querySelector( 'a.block-editor-inserter__manage-reusable-blocks' );
+			if ( hyperlink ) {
+				hyperlink.href = manageReusableBlocksUrl;
+				hyperlink.target = '_top';
+			}
+		}
+	} );
+
+	const sidebarsObserver = new window.MutationObserver( ( mutations ) => {
+		for ( const record of mutations ) {
+			for ( const node of record.addedNodes ) {
+				if (
+					createNewPostUrl &&
```

Comment:
`nb:` It might be worthwhile to turn the if conditional clause into a shared, local variable ( the naming can be adjusted). It can be used here and on line 740. I feel like it adds clarity to what the conditional is trying to capture.

```javascript
const shouldReplaceSidebarLinks = createNewPostUrl && 
    ( node.classList.contains( 'interface-interface-skeleton__sidebar' ) || // Site editor
	node.classList.contains( 'edit-post-sidebar' ) ) // Post editor
```

The same can probably be done for he manage reusable blocks else if clause.